tool/colors/update_colors.dart

```diff
@@ -19,7 +19,9 @@ const String generatedFilesPath = 'tool/colors/generated';
 
 Future<void> main(List<String> args) async {
   // Verify that we're running from the project root.
-  if (path.basename(Directory.current.path) != 'tools_metadata') {
+  if (!File(path.join(Directory.current.path, 'pubspec.yaml'))
+      .readAsStringSync()
+      .startsWith('name: tool_metadata')) {
```

Comment:
It's quite simple because my fork is `flutter_tool_metadata` 😄. For the most precise check, we can use `loadYaml` and then find the specific field to avoid comment things.

Reply:
> It's quite simple because my fork is `flutter_tools_metadata` 😄

Ah! I hadn't checked and had assumed this was a moo-repo and that folder name was inside the repo 🤦‍♂️

Maybe we could be slightly more lax with `contains()`?

I also wonder if you're in the wrong folder would this throw a `FileSystemException` rather than just failing the check?

Reply:
I've created a new `fromTheProjectRoot` method and applied it to generators.